custom_components/ha_washdata/profile_store.py

```
@@ -2286,6 +2338,71 @@ async def async_delete_maintenance_event(self, event_id: str) -> bool:
+        self, name: str, values: dict[str, Any]
+    ) -> JSONDict:
+        """Create or overwrite a named Playground preset and persist it.
+
+        ``values`` must already be sanitized by ``playground.sanitize_setting_values``
+        (the store deliberately does not import the playground module - that would
+        be a cycle). Raises ``ValueError`` for an empty name, an empty value map, or
+        when the per-device preset cap is reached by a NEW name.
+        """
+        name = (name or "").strip()[:PLAYGROUND_PRESET_NAME_MAX]
+        if not name:
+            raise ValueError("Preset name is required")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)